### PR TITLE
Docs: add project roadmap

### PR DESCRIPTION
**Pull request by lizrice**
A project roadmap helps users understand the direction of the project, and it helps potential contributors see where they might be able to get involved. (Also see [CNCF recommendations](https://contribute.cncf.io/maintainers/community/contributor-growth-framework/motivation/#honest-and-clear-communication) for why communicating a roadmap publicly is a good idea). 

We want readers to immediately see that Cilium is essentially stable and widely used in production (hence the “feature status” table), and that there is active and ongoing development. 

It’s not intended to be a comprehensive list of absolutely everything that we're working on, or a replacement for GitHub - it’s more of a general statement of direction. This will be a living document that needs to be updated after each major release.

### CODE REVIEW

File: Documentation/community/roadmap.rst
Comment (qmonnet) [inline Documentation/community/roadmap.rst]:
```suggestion
broader community. You'll also find here some pointers on how you can
:ref:`influence the roadmap<rm-influence>`. 
```

File: Documentation/community/roadmap.rst
Comment (qmonnet) [inline Documentation/community/roadmap.rst]:
```suggestion

```

File: Documentation/community/roadmap.rst
Comment (qmonnet) [inline Documentation/community/roadmap.rst]: Can you use consistent casing on the titles, please (here and for other titles)? It seems that most of them use title case.

File: Documentation/community/roadmap.rst
Comment (qmonnet) [inline Documentation/community/roadmap.rst:49]:
```suggestion
please refer to `issues on GitHub <GitHub issues>`_. 
```

File: Documentation/community/roadmap.rst
Comment (qmonnet) [inline Documentation/community/roadmap.rst]: I'll have a look at it :)

File: Documentation/community/roadmap.rst
Comment (qmonnet) [inline Documentation/community/roadmap.rst]:
```suggestion
focus currently include:
```

File: Documentation/community/roadmap.rst
Comment (qmonnet) [inline Documentation/community/roadmap.rst]:
```suggestion

```

**Review by qmonnet — CHANGES_REQUESTED** [review]
Looks good overall, please find some minor formatting nits inline below.

I'll look at the Maglev title, I'll also look at the "two misspelled" words reported but not displayed by the CI (I've changed things here recently, probably my fault if it doesn't show as expected).

Please address the other CI report: update the CODEOWNERS file with a new entry for the roadmap document, and update the ones for the files that were moved.

File: Documentation/community/roadmap.rst
Comment (qmonnet) [reply Documentation/community/roadmap.rst:49]: Although this one seems to be working fine as is, so not sure if this change is required or if I missed something.

**Comment by qmonnet** [comment]
> I'll look at the Maglev title

You probably tried to use `maglev` as a target name, just as I did. This fails with the following error:

```
./gettingstarted/kubeproxy-free.rst:11: (WARNING/2) Duplicate explicit target name: "maglev"
```

This is because the same document latest defines a link target like this:

```
`Maglev <https://storage.googleapis.com/pub-tools-public-publication-data/pdf/44824.pdf>`_
```

and this creates a conflict. This can be solved by using a different target name, or by making the above link (to the PDF) _anonymous_, by adding a second underscore. Yeah, RST is fun!

<details><summary>Fix for your PR</summary>

```diff
diff --git a/Documentation/community/roadmap.rst b/Documentation/community/roadmap.rst
index 986c7e01ccbe..275bbc30328a 100644
--- a/Documentation/community/roadmap.rst
+++ b/Documentation/community/roadmap.rst
@@ -141,7 +141,7 @@ following features to graduate to stable:
 * :ref:`Local Redirect Policy<local-redirect-policy>`
 * :ref:`Egress Gateway<egress-gateway>`
 * :ref:`CiliumEndpointSlice<gsg_ces>`
-* `Maglev Consistent Hashing`_
+* :ref:`Maglev Consistent Hashing<maglev>`
 
 .. _rm-hubble-observability:
 
@@ -214,5 +214,3 @@ anything other than trivial fixes.
 .. _good-first-issue: https://github.com/cilium/cilium/labels/good-first-issue
 .. _slack: https://cilium.io/slack
 .. _enterprise: https://cilium.io/enterprise
-
-.. _Maglev Consistent Hashing: ../../gettingstarted/kubeproxy-free.html#maglev-consistent-hashing-beta
\ No newline at end of file
diff --git a/Documentation/gettingstarted/kubeproxy-free.rst b/Documentation/gettingstarted/kubeproxy-free.rst
index 44ade58ecbac..1390de25bbb5 100644
--- a/Documentation/gettingstarted/kubeproxy-free.rst
+++ b/Documentation/gettingstarted/kubeproxy-free.rst
@@ -315,6 +315,8 @@ be lost on its path to the service endpoint.
   or :ref:`Hybrid<Hybrid Mode>` mode if only TCP-based services are exposed to the outside
   world for the latter.
 
+.. _maglev:
+
 Maglev Consistent Hashing (Beta)
 ********************************
 
@@ -348,7 +350,7 @@ There are two more Maglev-specific configuration settings: ``maglev.tableSize``
 and ``maglev.hashSeed``.
 
 ``maglev.tableSize`` specifies the size of the Maglev lookup table for each single service.
-`Maglev <https://storage.googleapis.com/pub-tools-public-publication-data/pdf/44824.pdf>`_
+`Maglev <https://storage.googleapis.com/pub-tools-public-publication-data/pdf/44824.pdf>`__
 recommends the table size (``M``) to be significantly larger than the number of maximum expected
 backends (``N``). In practice that means that ``M`` should be larger than ``100 * N`` in
 order to guarantee the property of at most 1% difference in the reassignments on backend
```
</details>

---

> I'll also look at the "two misspelled" words reported but not displayed by the CI

Here are the two words that Sphinx complains about:

```
* Documentation/community/roadmap.rst:88: (conformant)  100% upstream conformant                                                                                                                                 
* Documentation/community/roadmap.rst:93: (servicemesh)  Using Kubernetes as servicemesh control plane
```

<details><summary>Not displayed because under a directory, when we try to process files (I'll send a PR)</summary>

``` diff
diff --git a/Documentation/check-build.sh b/Documentation/check-build.sh
index 858512a88281..90712081063c 100755
--- a/Documentation/check-build.sh
+++ b/Documentation/check-build.sh
@@ -36,7 +36,7 @@ has_spelling_errors() {
 describe_spelling_errors() {
     printf "\nPlease fix the following spelling mistakes:\n"
     # show file paths relative to repo root
-    sed 's/^/* Documentation\//' "${spelldir}"/*
+    find "${spelldir}" -type f -print0 | xargs -0 sed 's/^/* Documentation\//'
 }
 
 hint_about_wordlist_update() {
```
</details>

File: Documentation/community/roadmap.rst
Comment (qmonnet) [reply Documentation/community/roadmap.rst]: See my comment below https://github.com/cilium/cilium/pull/19540#issuecomment-1106651401

File: Documentation/community/roadmap.rst
Comment (borkmann) [inline Documentation/community/roadmap.rst]: bandwidth manager and egress GW (@jibi) will be stable in 1.12, so they could probably go to your major feature status? (Maglev is actually used in prod, too. We should move it out of beta from the doc. [edit] PR is here: https://github.com/cilium/cilium/pull/19541)

File: Documentation/community/roadmap.rst
Comment (sayboras) [inline Documentation/community/roadmap.rst:32]: It will be great to have something like Stable from version 1.x.x, something like upstream k8s docs has.

Also, for service mesh, is the plan to mark it as stable from 1.12?

File: Documentation/community/roadmap.rst
Comment (lizrice) [reply Documentation/community/roadmap.rst]: This will (I hope) be live before 1.12 is released, so I'll do an update after that. But are we saying that Maglev is stable in 1.11?

File: Documentation/community/roadmap.rst
Comment (lizrice) [reply Documentation/community/roadmap.rst:32]: I like the idea, but let's add this as an improvement on a future update so that we don't hold this up

File: Documentation/community/roadmap.rst
Comment (qmonnet) [reply Documentation/community/roadmap.rst]: Maglev-related PR: https://github.com/cilium/cilium/pull/19541, hopefully stable in 1.12

**Comment by qmonnet** [comment]
@lizrice here's a diff that should fix CI with regards to the CODEOWNERS to update, and the spelling reports (although `servicemesh` appears elsewhere in the docs as `service mesh`, two words, so maybe switch to that?).

<details><summary>diff</summary>

``` diff
diff --git a/CODEOWNERS b/CODEOWNERS
index eebfcf48efd2..266121db7089 100644
--- a/CODEOWNERS
+++ b/CODEOWNERS
@@ -99,7 +99,10 @@ Makefile* @cilium/build
 /Documentation/check-examples.sh @cilium/docs-structure
 /Documentation/check-helmvalues.sh @cilium/docs-structure
 /Documentation/cmdref/
-/Documentation/community.rst @cilium/contributing
+/Documentation/community/community.rst @cilium/contributing
+/Documentation/community/governance/ @cilium/contributing
+/Documentation/community/governance/commit_access.rst @cilium/contributing @cilium/docs-structure
+/Documentation/community/roadmap.rst @cilium/contributing @cilium/docs-structure
 /Documentation/concepts/clustermesh @cilium/sig-clustermesh @cilium/docs-structure
 /Documentation/concepts/ebpf/ @cilium/bpf @cilium/docs-structure
 /Documentation/concepts/index.rst @cilium/docs-structure
@@ -113,7 +116,6 @@ Makefile* @cilium/build
 /Documentation/conf.py @cilium/docs-structure
 /Documentation/configuration/index.rst @cilium/docs-structure
 /Documentation/contributing/ @cilium/contributing @cilium/docs-structure
-/Documentation/contributing/governance/commit_access.rst @cilium/contributing @cilium/docs-structure
 /Documentation/Dockerfile @cilium/docs-structure
 /Documentation/gettinghelp.rst @cilium/contributing @cilium/docs-structure
 /Documentation/gettingstarted/alibabacloud* @cilium/alibabacloud @cilium/docs-structure
diff --git a/Documentation/spelling_wordlist.txt b/Documentation/spelling_wordlist.txt
index 821d24f558a2..e144f7cc063b 100644
--- a/Documentation/spelling_wordlist.txt
+++ b/Documentation/spelling_wordlist.txt
@@ -269,6 +269,7 @@ configMap
 configMapKey
 configmap
 configs
+conformant
 conntrack
 conntrackGCInterval
 containerRuntime
@@ -812,6 +813,7 @@ serviceAccounts
 serviceAnnotations
 serviceMonitor
 servicePort
+servicemesh
 sessionAffinity
 setrlimit
 setupinterface
```
</details>

File: Documentation/community/roadmap.rst
Comment (lizrice) [reply Documentation/community/roadmap.rst]: done

**Review by qmonnet — APPROVED** [review]
Thanks a lot!

**Review by sayboras — APPROVED** [review]

**Review by joestringer — APPROVED** [review]

**Comment by xmulligan** [comment]
Should we add a link to this in the readme?

**Comment by qmonnet** [comment]
Speaking of which, the link to the Governance documents [on the README](https://github.com/cilium/cilium#governance) is now broken.

**Comment by xmulligan** [comment]
I've added it here https://github.com/cilium/cilium/pull/19615

**Pull request merged**
